File: backend/satisfactory-blueprint/satisfactory-blueprint/src/main/java/com/satisfactory/blueprint/dto/ImageUploadRequest.java
package com.satisfactory.blueprint.dto;

import com.satisfactory.blueprint.entity.enums.OwnerType;
import lombok.Data;

import java.util.UUID;

@Data
public class ImageUploadRequest {
    private UUID id;
    private String contentType;
    private String data;
    private UUID oldImageId;
    private OwnerType ownerType;
    private Long ownerId;

}

File: frontend/satisfactory-blueprint-designer/src/types/imageUploadRequest.ts
// src/types/imageUploadRequest.ts

import { type OwnerType } from './enums';

/**
 * Mirrors com.satisfactory.blueprint.dto.ImageUploadRequest
 */
export interface ImageUploadRequest {
  /** UUID of the image to reuse; omit or null when uploading a new file */
  id?: string;
  /** UUID of the previous image to unlink (if any) */
  oldImageId?: string;
  /** Which entity this image belongs to */
  ownerType: OwnerType;
  /** Primary key of that entity */
  ownerId: number;
  /** MIME type of the new image (required when `data` is provided) */
  contentType?: string;
  /** Base64-encoded image data for uploads; omit when reusing an existing image */
  data?: string;
}
